src/opera_utils/_utils.py

```diff
@@ -153,3 +161,238 @@ def scratch_directory(
+
+    Parameters
+    ----------
+    tropo_delay_cube : dict
+        Dictionary containing tropospheric delay data.
+    grid : dict
+        Dictionary containing grid information.
+    geo_files : dict[str, Path]
+        Dictionary containing paths to geospatial files.
+
+    Returns
+    -------
+    dict
+        Dictionary containing computed 2D delay.
+    """
+    dem_file = geo_files["height"]
+
+    ysize, xsize = grid["shape"]
+    x_origin, x_res, _, y_origin, _, y_res = grid["geotransform"]
+
+    gt = grid["geotransform"]
+
+    x = 0
+    y = 0
+    left = gt[0] + x * gt[1] + y * gt[2]
+    top = gt[3] + x * gt[4] + y * gt[5]
+
+    x = xsize
+    y = ysize
+
+    right = gt[0] + x * gt[1] + y * gt[2]
+    bottom = gt[3] + x * gt[4] + y * gt[5]
+
+    bounds = (left, bottom, right, top)
+
+    options = gdal.WarpOptions(
+        dstSRS=grid["crs"],
+        format="MEM",
+        xRes=x_res,
+        yRes=y_res,
+        outputBounds=bounds,
+        outputBoundsSRS=grid["crs"],
+        resampleAlg="near",
+    )
+    target_ds = gdal.Warp(
+        os.path.abspath(fspath(dem_file) + ".temp"),
+        os.path.abspath(fspath(dem_file)),
+        options=options,
+    )
+
+    dem = target_ds.ReadAsArray()
+
+    los_east = load_gdal(geo_files["los_east"])
+    los_north = load_gdal(geo_files["los_north"])
+    los_up = 1 - los_east**2 - los_north**2
+
+    mask = los_east > 0
+
+    # Make the x/y arrays
+    # Note that these are the center of the pixels, whereas the GeoTransform
+    # is the upper left corner of the top left pixel.
+    y = np.arange(y_origin, y_origin + y_res * ysize, y_res)
+    x = np.arange(x_origin, x_origin + x_res * xsize, x_res)
+
+    yv, xv = np.meshgrid(y, x, indexing="ij")
+
+    delay_2d = {}
+    for delay_type in tropo_delay_cube.keys():
+        if delay_type not in ["x", "y", "z"]:
+            # tropo_delay_datacube_masked = np.ma.masked_invalid(tropo_delay_cube[delay_type])
+
+            tropo_delay_interpolator = RegularGridInterpolator(
+                (grid["height_levels"], grid["ycoord"], grid["xcoord"]),
+                tropo_delay_cube[delay_type],
+                method="linear",
+                bounds_error=False,
+            )
+
+            tropo_delay_2d = np.zeros(dem.shape, dtype=np.float32)
+
+            nline = 100
+            for i in range(0, dem.shape[1], 100):
+                if i + 100 > dem.shape[0]:
+                    nline = dem.shape[0] - i
+                pnts = np.stack(
+                    (
+                        dem[i : i + 100, :].flatten(),
+                        yv[i : i + 100, :].flatten(),
+                        xv[i : i + 100, :].flatten(),
+                    ),
+                    axis=-1,
+                )
+                tropo_delay_2d[i : i + 100, :] = tropo_delay_interpolator(pnts).reshape(
+                    nline, dem.shape[1]
+                )
+
+            out_delay_type = delay_type.replace("Zenith", "LOS")
```

Comment:
Similar idea for this output type as the input types- right now, we're relying on the `dict` keys being strings which are correctly formatted, and then the user parses them to know all the relevant info about the `delay_2d` data. It looks like we're making this key like this
```
        tropo_delay_product_name = f'tropoDelay_pyaps_{model_type}_Zenith_{tropo_delay_product}'
        troposphere_delay_datacube[tropo_delay_product_name]  = tropo_delay_datacube_masked
```

So the information contained in the key is:
- the value is tropospheric delay (which we could probably indicate by a dataclass name like `TroposphericDelay`
- the model type- i think these are from choices `(ERA5, HRES)`, so we can [make that an `Enum`](https://docs.python.org/3/library/enum.html), or in the dataclass make it [a `Literal` type](https://docs.python.org/3/library/typing.html#typing.Literal), like `Literal['ERA5', 'HRES']` to indicate it can only be one of those
- the direction ("Zenith" or "LOS"). again, it could be an enum
- the product_type: `['wet', 'hydrostatic', 'comb']`
It seems useful to break that out into a dataclass with those 4 fields, rather than have a compound string which we have to parse later on